packages/markup-frontstage/src/services/SavedView.ts

```diff
@@ -71,9 +82,70 @@ export const parseFromEmphasizeElements = (
 };
 
 /**
- * Creates saved view data
+ * Convert given object to string or undefined if it's undefined.
+ * @param args name of the argument object.
+ */
+export const jsonStringify = <T>(args: T): string | undefined => {
+  return args !== undefined ? JSON.stringify(args) : undefined;
+};
+
+/**
+ * Creates 2D view props that is common to both sheet and drawings.
+ * @param vp the name of the view port.
+ */
+const create2dViewProps = (vp: Viewport): SavedViewData => {
+  const viewState = vp.view as ViewState2d;
+  const categorySelectorProps = viewState.categorySelector?.toJSON();
+  const viewDefinitionProps = viewState.toJSON();
+  const displayStyleProps = viewState.displayStyle?.toJSON();
+  const emphasizedElementsProps = EmphasizeElements.get(vp)?.toJSON(vp);
+
+  return {
+    categorySelectorProps: jsonStringify(categorySelectorProps),
+    displayStyleProps: jsonStringify(displayStyleProps),
+    emphasizedElementsProps: jsonStringify(emphasizedElementsProps),
+    is2d: true,
+    viewDefinitionProps: jsonStringify(viewDefinitionProps),
+  };
+};
+
+/**
+ * Created 2D saved view data either sheet or drawing on the basis of view state type.
+ * @param vp name of the view port.
+ * @param viewStateType name fo the view state type.
+ */
+export const create2DSavedView = (
+  vp: Viewport,
+  viewStateType: string
+): SavedViewData => {
+  if (vp.view === undefined) {
+    throw new Error("Invalid viewport");
+  }
+  const common2dViewProps = create2dViewProps(vp);
+  if (viewStateType === DrawingViewState.name) {
+    return common2dViewProps;
+  }
+  if (viewStateType === SheetViewState.name) {
+    const viewState = vp.view as SheetViewState;
+    const sheetSize = viewState.sheetSize;
+    const sheetProps: SheetProps = {
+      width: sheetSize.x,
+      height: sheetSize.y,
+      model: viewState.model,
+      classFullName: SheetViewState.classFullName,
+      code: Code.createEmpty(),
+    };
+    const common2dViewProps = create2dViewProps(vp);
```

Comment:
Do we need a new const here? Couldn't we just use the common2dViewProps that was already declared?

Reply:
good catch, removed the line 👍